kafka-admin/src/main/resources/graphql-schema/kafka-admin.graphql

```diff
@@ -1,11 +1,67 @@
-extend type Query {
-    topic(topicName: String): Topic
-    topicList(filter: String): [Topic]
+type TopicDescription {
+    #id: ID
+    name: String
+
+    """@transient"""
+    partitions: [Partitions]
+    config: TopicConfig
+}
+
+type Partitions {
+    partition: Int
+    replicas: [Replicas]
+}
+
+type Replicas {
+    id: String
+    inSync: Boolean
 }
 
-type Topic {
+type TopicOnlyName {
+    #id: ID
     name: String
+}
+
+type TopicConfig {
+    partitionCount: Int
+    replicationFactor: Int
     isInternal: Boolean
+    # config
+    minInsyncReplicas: Int
+    pairs: [ConfigEntry]
+}
+
+type ConfigEntry {
+    key: String
+    value: String
+}
+
+input CreateOrMutateTopicConfigInput {
+    retentionDays: Int
+    minInSyncReplicas: Int
     partitionCount: Int
     replicationFactor: Int
+    isInternal: Boolean
+}
+
+input CreateTopicInput {
+    name: String
+    config: CreateOrMutateTopicConfigInput
+}
+
+input MutateTopicInput {
+    #id: ID!
+    name: String!
+    config: CreateOrMutateTopicConfigInput
+}
+
+extend type Mutation {
+    createTopic(input: CreateTopicInput!): TopicDescription
+    updateTopic(input: MutateTopicInput!): TopicDescription
+    deleteTopic(name: String!): String
+}
+
+extend type Query {
+    topicList(filter: String): [TopicOnlyName!]
```

Comment:
Introducing `scalar JSON` to your schema would be very good for todo items like filters etc. until structure of the filtering will be agreed upon.

Reply:
I am very beginner with these schemas. Can you be more descriptive, please?

Reply:
```suggestion
    topicList(filter: JSON): [TopicOnlyName!]
```

See:
https://github.com/graphql-java/graphql-java-extended-scalars